examples/vision/text_image_denoiser.py

```diff
@@ -0,0 +1,296 @@
+"""
+"""
+## Introduction
+
+This Example explains a simple **Text Image Denoiser for OCR** using U-Net based
+Architectutre. **Autoencoders** are mostly employed for **Image Restoration** problems
+but in recent times **U-Net** based architectures with **skip-connections** have gained
+popularity for several Image-to-Image tasks. We aim to solve the problem with simple
+**Pretrained U-Net as Encoder** and **Efficient Sub-Pixel CNN as Decoder**. The problem
+we focus here is enhancing the Document Images which are deteriorated through external
+degradations like blur, corrupt text blocks etc. We have followed up on few seminal
+papers which are presented below for reference. At the end of this tutorial user will
+gain clear understanding of building Custom Image Pre-Processors using Deep-Learning that
+helps us to mitigate real world OCR issues.
+
+**References:**
+- [Enhancing OCR Accuracy with Super
+Resolution](https://cdn.iiit.ac.in/cdn/cvit.iiit.ac.in/images/ConferencePapers/2018/ocr_Ankit_Lat_ICPR_2018.pdf)
+- [Improving the Perceptual Quality of Document
+Images Using Deep Neural
+Network](http://mile.ee.iisc.ac.in/publications/softCopy/DocumentAnalysis/ISNN_11page_65.pdf)
+"""
+
+"""
+## Additional Set-up
+"""
+
+# !pip install pybind11
+# !pip install fastwer
+# !pip install pytesseract
+# !sudo apt install tesseract-ocr
+
+import fastwer
+import pytesseract
+import numpy as np
+from glob import glob
+import tensorflow as tf
+from tensorflow import keras
+import matplotlib.pyplot as plt
+from tensorflow.keras import callbacks
+from tensorflow.keras.applications.resnet_v2 import ResNet50V2, preprocess_input
+
+"""
+## Dataset
+We have created a synthetic document dataset that depicts several real world document
+noises; we have used Blurs and Morphological filters to reconstruct real world noises.
+"""
+
+"""shell
+!gdown https://drive.google.com/uc?id=1jQfHJUUQcpktQcjIzitvb4SldKwqcAMr
+!unzip -q data.zip
+"""
+
+"""
+## Dataset Pipeline
+"""
+
+train_source = sorted(glob("data/source/*.jpg"))[:-80]
+train_targets = sorted(glob("data/target/*.jpg"))[:-80]
+
+val_source = sorted(glob("data/source/*.jpg"))[-80:]
+val_targets = sorted(glob("data/target/*.jpg"))[-80:]
+
+HEIGHT, WIDTH = 352, 608
+BATCH_SIZE = 4
+BUFFER_SIZE = int(len(train_source) * 0.2)
+
+
+def resize_with_pad(image, height, width):
+    return tf.image.resize_with_pad(
+        image=image, target_height=height, target_width=width
+    )
+
+
+def preprocess(image):
+    image = tf.io.decode_png(tf.io.read_file(image), channels=3)
+    image = resize_with_pad(image, height=HEIGHT, width=WIDTH)
+    image = preprocess_input(image)
+    return image
+
+
+def data_preprocess(source, target):
+    source_image = preprocess(source)
+    target_image = preprocess(target)
+    return source_image, target_image
+
+
+def denormalize(array):
+    array += 1
+    array *= 127.5
+    return array
+
+
+train_pipe = tf.data.Dataset.from_tensor_slices((train_source, train_targets))
```

Comment:
@anish9 `training_set` (not `trian_pipe`)

Reply:
@innat  the following final Variable name is given as per your guideline, 

```
train_pipe = tf.data.Dataset.from_tensor_slices((train_source,train_targets))
train_pipe = train_pipe.map(data_preprocess,tf.data.AUTOTUNE).shuffle(BUFFER_SIZE)
train_set = train_pipe.batch(BATCH_SIZE).repeat()
```